cirq-core/cirq/transformers/transformer_primitives.py

```diff
@@ -104,6 +104,89 @@ def map_moments(
     )
 
 
+def _map_operations_impl(
```

Comment:
I think this function is complex enough to deserve a docstring explaining its logic.

Reply:
added a detailed docstring.